server/Makefile.am

```diff
@@ -113,7 +114,7 @@ seaf_server_SOURCES = \
 
 seaf_server_LDADD = @CCNET_LIBS@ \
 	$(top_builddir)/lib/libseafile_common.la \
-	@GLIB2_LIBS@ @GOBJECT_LIBS@ @SSL_LIBS@ @LIB_RT@ @LIB_UUID@ -lsqlite3 @LIBEVENT_LIBS@ -levhtp \
+	@GLIB2_LIBS@ @GOBJECT_LIBS@ @SSL_LIBS@ @LIB_RT@ @LIB_UUID@ -lsqlite3 @LIBEVENT_LIBS@ @EVHTP_LIBS@ \
```

Comment:
```libevhtp``` depends on ```libonig```. ```libonig``` declares symbols which overload the ones used in ```glibc``` (```regcomp```, ```regexp_t```, etc.). As is, this line will make ```seaf-server``` use ```glibc``` symbols instead of ```libonig```. This will lead to stack smashing because of the different size of ```regexp_t```. Adding ```-lonig``` before works around the problem because then ```libonig``` symbols will take precedence over ```glibc``` ones. But this is overlinking which is usually discouraged. However, I don't know of a better fix.

Reply:
The cleanest way is to rename regex_t in libevhtp, I believe. Since C has no namespacing support, the common way is to emulate namespacing by adding some prefix to the exported names of structures and functions (for instance regex_t -> evhtp_regex_t).

Reply:
regex_t is redeclared in libonig, not libevhtp.